genesis/process/shardGenesisBlockCreator.go

```diff
@@ -429,14 +446,18 @@ func createProcessorsForShardGenesisBlock(arg ArgsGenesisBlockCreator, enableEpo
 		ESDTTransferParser:  esdtTransferParser,
 		BuiltInFunctions:    argsHook.BuiltInFunctions,
 		Hasher:              arg.Core.Hasher(),
-		PubKeyConverter:     arg.Core.AddressPubKeyConverter(),
-	}
-	vmFactoryImpl, err := shard.NewVMContainerFactory(argsNewVMFactory)
-	if err != nil {
-		return nil, err
-	}
-
-	vmContainer, err := vmFactoryImpl.Create()
+		PubkeyConv:          arg.Core.AddressPubKeyConverter(),
+		Economics:           arg.Economics,
```

Comment:
are all the newly added properties needed in the VMContainerFactory?

Reply:
Yes, needed for sovereign chain, since it also needs to create a meta container